### DISCUSSION TITLE
Starting server process programmatically

### DISCUSSION THREAD

**Discussion by msurdi** [vercel/next.js#22127] [Help]
Hi!

I'm developing a command line application that will start a Next application. Right now I'm resorting to doing things such as `exec("npm start")`, etc. 

Is this the best way to do this?

The only other way I can think of right now is by using a custom server, but that's also disencouraged as I will be missing some of the optimizations according to the docs.

Thanks!!

**Comment by mkrtchian** [comment]
You could use [pm2](https://pm2.keymetrics.io/) to start your next application programmatically in the background and manage it, [like described here](https://github.com/vercel/next.js/discussions/20943#discussioncomment-273162).

**Reply by msurdi** [reply]
Sorry if I didn't explain myself clearly enough.
I don't want to start my application as a service. I'm creating a CLI application, which you can install and run on your laptop, with several subcommands, one of which would be `myapp serve` and this one will start the Next.js application.

Think of it as the next dev command itself, just that it is my custom command which under the hood starts Next.

Thanks for your help.

**Comment by saifsmailbox98** [comment]
@msurdi Did you find a solution to this?

**Reply by msurdi** [reply]
I've moved away from Next since I needed this. Sorry for not being of any help.

**Comment by yukulelix** [comment]
Keystone is doing this under the hood. You can check the code here : https://github.com/keystonejs/keystone/blob/main/packages/core/src/scripts/dev.ts
